pilot/pkg/xds/ads.go

```diff
@@ -269,6 +269,10 @@ func (s *DiscoveryServer) Stream(stream DiscoveryStream) error {
 		return status.Errorf(codes.ResourceExhausted, "request rate limit exceeded: %v", err)
 	}
 
+	// TODO: get WI from peerAddr and use it to validate auth and (if secure net)
+	// - populate metadata from WI if missing
+	// - validate namespace and sensitive meta
```

Comment:
we already do this in s.authorize

Reply:
The second part, and only namespace AFAIK - but I didn't find it for the 15010 port.

Reply:
(part of a separate PR, a bit hard to keep them separated, I can remove from this )